Code/Tools/AssetProcessor/native/resourcecompiler/JobsModel.cpp

```diff
@@ -359,23 +360,30 @@ namespace AssetProcessor
             AZStd::unordered_map<QueueElementID, AZ::s64> historicalStats;
             auto statsFunction = [&historicalStats](AzToolsFramework::AssetDatabase::StatDatabaseEntry entry)
             {
-                AZStd::vector<AZStd::size_t> delimiterPos;
-                delimiterPos.reserve(3);
-                for (size_t pos = 0;
-                     (pos = entry.m_statName.find(',', delimiterPos.empty() ? 0 : (delimiterPos.back() + 1))) != AZStd::string::npos;)
+                AZStd::vector<AZStd::string> tokens;
+                AZ::StringFunc::Tokenize(entry.m_statName, tokens, ',');
+
+                if (tokens.size() == 4)
                 {
-                    delimiterPos.push_back(pos);
+                    QueueElementID elementId;
+                    elementId.SetInputAssetName(tokens[1].c_str());
+                    elementId.SetJobDescriptor(tokens[2].c_str());
+                    elementId.SetPlatform(tokens[3].c_str());
+                    historicalStats[elementId] = entry.m_statValue;
                 }
-                if (delimiterPos.size() != 3)
+                else
                 {
-                    return true;
+                    AZ_Warning(
+                        "AssetProcessor",
+                        false,
+                        "ProcessJob stat entry \"%s\" could not be parsed and will not be used. Expected %d tokens, but found %d. A wrong "
+                        "stat name may be used in Asset Processor code, or the asset database may be corrupted. If you keep encountering "
+                        "this warning, report an issue on GitHub with O3DE version number.",
+                        entry.m_statName.c_str(),
+                        tokens.size(),
+                        4); 
```

Comment:
I think these are backwards.  Also you might want to just make the 4 a static constexpr variable